code/modules/projectiles/guns/energy/hos.dm

```diff
@@ -0,0 +1,304 @@
+// -------------- HoS Modular Weapon System -------------
+// ---------- Code originally from VoreStation ----------
+/obj/item/gun/ballistic/revolver/mws
+	name = "MWS-01 'Big Iron'"
+	desc = "Modular Weapons System"
+
+	icon = 'icons/obj/guns/projectile.dmi'
+	icon_state = "mws"
+
+	fire_sound = 'sound/weapons/Taser.ogg'
+
+	mag_type = /obj/item/ammo_box/magazine/mws_mag
+	spawnwithmagazine = FALSE
+
+	recoil = 0
+
+	var/charge_sections = 6
+
+/obj/item/gun/ballistic/revolver/mws/afterattack(atom/target, mob/living/user, flag, params)
+	if(chambered && !chambered.BB) //if BB is null, i.e the shot has been fired...
+		var/obj/item/ammo_casing/mws_batt/shot = chambered
+		if(shot.shots_left >= 1)
+			shot.newshot() //Create a new shot
```

Comment:
ditto replace all this with a "new shot if empty and consume charge" proc on te microbattery